src/messagingsession/DefaultMessagingSession.ts

```diff
@@ -105,7 +107,32 @@ export default class DefaultMessagingSession implements MessagingSession {
   }
 
   private async startConnecting(reconnecting: boolean): Promise<void> {
-    const signedUrl = await this.prepareWebSocketUrl();
+    // reconnect needs to re-resolve endpoint url, which will also refresh credentials on client if they are expired
+    let endpointUrl = !reconnecting ? this.configuration.endpointUrl : undefined;
+    if (endpointUrl === undefined) {
+      try {
+        if (this.configuration.chimeClient.getMessagingSessionEndpoint instanceof Function) {
+          endpointUrl = (
+            await this.configuration.chimeClient.getMessagingSessionEndpoint().promise()
+          ).Endpoint.Url;
+        } else {
+          endpointUrl = (
+            await this.configuration.chimeClient.send(new GetMessagingSessionEndpointCommand({}))
+          ).Endpoint.Url;
+        }
+      } catch (e) {
```

Comment:
Would be good to add a log here on successful `endpointUrl` retrieval and same for error as well.

Reply:
thx will update.